packages/wonder-blocks-core/components/id-provider.md

```diff
@@ -0,0 +1,16 @@
+### Example: Usage with form fields
+
+This example allows you to generate an unique ID and make it available to associate the `<label>` and `<input>` elements. To see this example in action, check that `label[for]` and `input[id]` are using the same id.
+
+```jsx
+<View>
+    <IDProvider scope="field">
+        {(uniqueId) => (
+            <label htmlFor={uniqueId}>
+                Label with ID {uniqueId}:
+                <input type="text" id={uniqueId} />
+            </label>
+        )}
+    </IDProvider>
+</View>
+```
```

Comment:
I think another example where the id prop is used would also be helpful. Perhaps a scenario where we have a component that takes an optional `id` prop, which then gets passed internally to the `IDProvider`, which then either uses that `id` prop if it was set, but otherwise defaults to a unique ID. This would show why this component exists; the idea that it's for cases where you need an ID, so if one isn't given, it will enforce one.